akka-typed-tests/src/test/scala/akka/typed/scaladsl/adapter/AdapterSpec.scala

```diff
@@ -27,41 +27,43 @@ object AdapterSpec {
   }
 
   def typed1(ref: untyped.ActorRef, probe: ActorRef[String]): Behavior[String] =
-    Immutable(
-      onMessage = (ctx, msg) ⇒
-      msg match {
-        case "send" ⇒
-          val replyTo = ctx.self.toUntyped
-          ref.tell("ping", replyTo)
-          Same
-        case "pong" ⇒
-          probe ! "ok"
-          Same
-        case "actorOf" ⇒
-          val child = ctx.actorOf(untyped1)
-          child.tell("ping", ctx.self.toUntyped)
-          Same
-        case "watch" ⇒
-          ctx.watch(ref)
-          Same
-        case "supervise-stop" ⇒
-          val child = ctx.actorOf(untyped1)
-          ctx.watch(child)
-          child ! ThrowIt3
-          child.tell("ping", ctx.self.toUntyped)
+    Immutable[String] {
```

Comment:
some type inference is lost, but I don't know exactly why.
javac had same problem when I tried same thing in javadsl (and there this was a showstopper, see my other PR)